homeassistant/components/androidtv/media_player.py

```diff
@@ -324,8 +324,13 @@ def __init__(self, aftv, name, apps, turn_on_command,
         self._device = None
         self._device_properties = self.aftv.device_properties
         self._is_volume_muted = None
-        self._unique_id = 'androidtv-{}-{}'.format(
-            name, self._device_properties['serialno'])
+
+        if 'serialno' in self._device_properties:
+            self._unique_id = 'androidtv-{}'.format(
```

Comment:
also remove `androidtv-` from the unique ID. Just set it to `self._device_properties['serialno']`

Reply:
Done.